### DISCUSSION TITLE
Remove `var`, `let`, `alias`

### DISCUSSION THREAD

**Discussion by remorses** [modular/modular#1069] [Ideas]
I don't see much value in these keywords, the compiler already knows if an assignment is a declaration

Same thing for `alias`, automatically use alias if a variable is used at compile time

Also mutability checks already exist in functions arguments, adding it to the variable instantiation usually is superfluous

**Comment by cjohnson318** [comment]
A large part of programming is communicating intent to future programmers. Just because a compiler can figure something out, doesn't mean that you can figure the same thing out in a non-trivial amount of time.

**Reply by gryznar** [reply]
I like this explanation! Thanks :)

**Reply by remorses** [reply]
What if `var` was removed and we only keep `let`?

**Reply by ormeilu** [reply]
Those keywords helps to identify - when variable is declared and when you just modifying it

**Comment by lattner** [comment]
As above, these are needed for clarity in code and predictability, not just the compiler.  Further rationale is in the programmer manual
